NitroxServer/GameLogic/EscapePodManager.cs

```diff
@@ -38,70 +42,53 @@ public NitroxId AssignPlayerToEscapePod(ushort playerId, out Optional<EscapePodM
                 return escapePodsByPlayerId[playerId].Id;
             }
 
-            if (podForNextPlayer.IsFull())
+            if (IsPodFull(podForNextPlayer))
             {
                 newlyCreatedPod = Optional.Of(CreateNewEscapePod());
                 podForNextPlayer = newlyCreatedPod.Value;
             }
 
-            podForNextPlayer.AssignedPlayers.Add(playerId);
+            podForNextPlayer.Players.Add(playerId);
             escapePodsByPlayerId[playerId] = podForNextPlayer;
 
             return podForNextPlayer.Id;
         }
 
-        public List<EscapePodModel> GetEscapePods()
+        private EscapePodWorldEntity CreateNewEscapePod()
         {
-            return EscapePods.ToList();
-        }
+            EscapePodWorldEntity escapePod = new EscapePodWorldEntity(GetStartPosition(), new NitroxId(), null);
 
-        public void RepairEscapePod(NitroxId id)
-        {
-            EscapePodModel escapePod = EscapePods.Find(ep => ep.Id.Equals(id));
-            escapePod.Damaged = false;
-        }
+            escapePod.ChildEntities.Add(new PrefabPlaceholderEntity(new NitroxId(), new NitroxTechType("Radio"), escapePod.Id));
+            escapePod.ChildEntities.Add(new PrefabPlaceholderEntity(new NitroxId(), new NitroxTechType("MedicalCabinet"), escapePod.Id));
+            escapePod.ChildEntities.Add(new PrefabPlaceholderEntity(new NitroxId(), new NitroxTechType("Fabricator"), escapePod.Id));
+            escapePod.ChildEntities.Add(new PrefabPlaceholderEntity(new NitroxId(), new NitroxTechType("SmallStorage"), escapePod.Id));
 
-        public void RepairEscapePodRadio(NitroxId id)
-        {
-            EscapePodModel escapePod = EscapePods.Find(ep => ep.RadioId.Equals(id));
-            escapePod.RadioDamaged = false;
-        }
-
-        private EscapePodModel CreateNewEscapePod()
-        {
-            EscapePodModel escapePod = new EscapePodModel();
-            escapePod.InitEscapePodModel(new NitroxId(),
-                                         GetStartPosition(),
-                                         new NitroxId(),
-                                         new NitroxId(),
-                                         new NitroxId(),
-                                         new NitroxId(),
-                                         true,
-                                         true);
-
-            EscapePods.Add(escapePod);
+            entityRegistry.AddEntity(escapePod);
+            entityRegistry.AddEntities(escapePod.ChildEntities);
```

Comment:
Will this add the child entities twice being part of the escapePod already? If not might we want to add this into the AddEntity code so we don't need to add them both individualy or do you not think it worth it?

Reply:
This case will not add redundant entities because the escape pod is only broadcasted to the client.  I think there might be implications to adding the children as well in AddEntities.  Off the top, other code paths using it might start throwing on duplicates and I have to check to see if there are any reasons that they shouldn't be in the registry. Will consider in a follow-up PR though.